src/databricks/labs/ucx/code/queries.py

```diff
@@ -0,0 +1,59 @@
+from typing import Iterable
+
+import sqlglot
+from sqlglot.expressions import Table
+
+from databricks.labs.ucx.code.base import Fixer, Diagnostic, Position, Range, Severity, Linter, DiagnosticTag
+
+from databricks.labs.ucx.hive_metastore.table_migrate import Index
+
+
+class FromTable(Linter, Fixer):
+    def __init__(self, index: Index):
+        self._index = index
+
+    def lint(self, query: str) -> Iterable[Diagnostic]:
+        for statement in sqlglot.parse(query):
+            if not statement:
+                continue
+            for table in statement.find_all(Table):
+                catalog = self._catalog(table)
+                if catalog != 'hive_metastore':
+                    continue
+                if not self._index.is_upgraded(table.db, table.name):
+                    continue
+                dst = self._index.get(table.db, table.name)
+                yield Diagnostic(
+                    # see https://github.com/tobymao/sqlglot/issues/3159
+                    range=Range.make(0, 0, 0, 1024),
+                    code='table-migrate',
+                    source='databricks.labs.ucx',
+                    message=f"Table {table.db}.{table.name} is migrated to {dst.destination()} in Unity Catalog",
+                    severity=Severity.ERROR,
+                    tags=[DiagnosticTag.DEPRECATED],
+                )
+
+    @staticmethod
+    def _catalog(table):
+        catalog = table.catalog
+        if not catalog:
+            catalog = 'hive_metastore'
+        return catalog
```

Comment:
We can make this method affirmative as,
```suggestion
        catalog = table.catalog
        if catalog:
            return catalog
        return "hive_metastore"
```